.github/actions/grype/action.yaml

```diff
@@ -0,0 +1,49 @@
+name: grype
+description: 'Run Grype on image'
+inputs:
+  image:
+    description: 'Image name'
+    required: true
+  registry:
+    description: 'Registry to login to pull image, e.g. "ghcr.io" for GHCR, leave empty if image is public'
+    required: false
+    default: ''
+  repo_owner:
+    description: 'Name of repository owner, e.g. "github.repository_owner" for ghcr.io'
+    required: false
+  repo_token:
+    description: 'Access token for repository owner, e.g. "secrets.GITHUB_TOKEN" for ghcr.io'
+    required: false
+  output:
+    description: 'Grype output either "sarif" (GITHUB_TOKEN with security-events:write) or print results as "table" and fail on error'
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Login with registry
+      if: inputs.registry != ''
+      uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a # v2.1.0
+      with:
+        registry: ${{ inputs.registry }}
+        username: ${{ inputs.repo_owner }}
+        password: ${{ inputs.repo_token }}
+    - name: Scan
+      if: inputs.output == 'table'
+      uses: anchore/scan-action@v3
```

Comment:
sha pinning?

Reply:
correct, I missed that. Will however at some point add the sha pins for all external actions. Still fixing here as well